java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/wiki/WikiDatabasePreparation.java

```diff
@@ -353,7 +389,411 @@ public static String removeMacroBlocks(StringBuilder text, Map<WikivoyageTemplat
+				date = dateValue.split(" ")[0].trim();
+			}
+		}
+		date = date.replaceAll("\\[+|]+", "");
+		return date;
+	}
+	
+	/**
+	 * Examples:
+	 * |description={{uk|1=Kyiv Pechersk Lavra}} => Kyiv Pechersk Lavra
+	 * |description=Some text description => Some text description
+	 */
+	private static Map<String, String> parseDescription(String line) throws IOException {
+		Map<String, String> result = new HashMap<>();
+		String descriptionBlock = line;
+		
+		while (descriptionBlock.contains("{{") && descriptionBlock.contains("}}")) {
+			// Find the start of the language
+			int langStart = descriptionBlock.indexOf("{{") + 2;
+			int langEnd = descriptionBlock.indexOf("|", langStart); // Look for the first "|"
+			if (langEnd == -1) {
+				break;  // If no "|", exit loop
+			}
+			
+			String lang = descriptionBlock.substring(langStart, langEnd).trim();
+			if (lang.length() > 5) {
+				break;  // If the language code is too long, exit loop
+			}
+			
+			// Look for the description part
+			int descStart = descriptionBlock.indexOf("|1=", langEnd) != -1
+					? descriptionBlock.indexOf("|1=", langEnd) + 3  // If "|1=" exists, use it
+					: langEnd + 1;  // Otherwise, start right after the first "|"
+			
+			// Count braces to correctly finish the language block
+			int openBraces = 1;  // We are already inside the first {{
+			int currentIndex = descStart;
+			
+			while (openBraces > 0 && currentIndex < descriptionBlock.length()) {
+				if (descriptionBlock.startsWith("{{", currentIndex)) {
+					openBraces++;
+				} else if (descriptionBlock.startsWith("}}", currentIndex)) {
+					openBraces--;
+				}
+				currentIndex++;
+			}
+			
+			if (openBraces == 0) {
+				String description = getDescString(descriptionBlock, descStart, currentIndex);
+				description = description.replaceAll("\\[+|]+", "");
+				result.put(lang, description);
+				// Move to the next block
+				descriptionBlock = descriptionBlock.substring(currentIndex).trim();
+			} else {
+				break;  // If closing brace is not found, exit
+			}
+		}
+		if (result.isEmpty()) {
+			// If no language block was found, use the whole description as the default language
+			String description = descriptionBlock.replaceFirst("(?i).*description\\s*=\\s*", "").trim();
+			if (description.startsWith("{{")) {
+				return result;  // skip it
+			}
+			final WikiModel wikiModel = new WikiModel("", "");
+			StringBuilder plainStr = new StringBuilder(wikiModel.render(new PlainTextConverter(true), description));
+			List<String> links = parseLinks(description);
+			if (!links.isEmpty()) {
+				plainStr.append("\n\nLinks:");
+				for (int i = 0; i < links.size(); i++) {
+					plainStr.append("\n[").append(i + 1).append("] ").append(links.get(i));
+				}
+			}
+			
+			result.put(DEFAULT_LANG, plainStr.toString());
+		}
+		
+		return result;
+	}
+	
+	private static List<String> parseLinks(String description) {
+		List<String> links = new ArrayList<>();
+		String regex = "\\[(http\\S+)\\s([^]]+)]";
```

Comment:
Is http**s**: parsing supported? `https? httpx? http/`? Additionally, weird part of regexp [^]] should be clarified. It is better to refactor this regexp from scratch.